open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -90,14 +90,25 @@ def get_available_docker_host(self, req_count, hackathon):
         # it's more reasonable to launch VM when the existed ones are almost used up.
         # The new-created VM must run 'cloudvm service by default(either cloud-init or python remote ssh)
         # todo the VM public/private IP will change after reboot, need sync the IP in db with azure in this case
+        service = Service(azure_key_id=azure_key_id)
         for docker_host in vms:
             if self.hosted_docker.ping(docker_host):
-                return docker_host
+                service_name = docker_host.public_dns.split(".")[0]
+                deployments = service.get_hosted_service_properties(service_name, detail=True).deployments
```

Comment:
One more thing, the code should work on local environment as well where `local` means our local development environment and it doesn't connect to azure at all.